crates/fluvio/src/config/cluster.rs

```diff
@@ -29,6 +29,9 @@ pub struct FluvioConfig {
     #[serde(default)]
     pub tls: TlsPolicy,
 
+    /// Cluster custom metadata
+    pub metadata: Option<toml::Value>,
```

Comment:
Since we are supporting multiple annotations, this should be `pub metadata: Vec<toml:Value>`

Reply:
it already supports multiple annotations

`metadata` will map to a `toml::Value::Table` which is basically a `BTreeMap`, so this example:

```toml
[cluster.name.metadata.one]
key = "value"

[cluster.name.metadata.two]
foo = "bar"
baz = 2
```

will deserialize (roughly) into:

```rust
metadata: BTreeMap {
    "one": BTreeMap { "key": "value" },
    "two": BTreeMap { "foo": "bar", "baz": 2 },
}
```

Reply:
Then it should be `pub metadata: Option<Table>` , which make it more strict.  Or it can be `pub metadata: Table` since Table is collection so no need for option

Reply:
I made the metadata private and made it a `Table`